### ISSUE TITLE
Add an argument to allow checks even if is_active is false?

### ISSUE THREAD

**Issue by ses4j** [django-guardian/django-guardian#343]
I want to give and check permissions to users that are created but not yet `is_active`.  Of course, all `has_perm` and `get_perms` checks are returning False, and I can't get around it.  Possible to add an argument to circumvent this check?  Any reason why not?

**Comment by brianmay** [comment]
What is the use case for this?

**Comment by brianmay** [comment]
What function are you talking about here? `get_objects_for_user` maybe?

This is already a very complicated function. Complexity means bugs like #327.

Don't like to add more arguments, especially if there is no use case. Maybe we should be thinking of ways of simplifying (if that is indeed possible) this code.

**Comment by ses4j** [comment]
Our use case was this: someone "shares" to a user that has no account yet.  We create the (inactive) user based on provided email, and set him up with the permissions given by the share.  That works so far (the library does allow SETTING perms on inactive users), but all queries that then check for the permission return false, which is very confusing.

**Comment by fgaudin** [comment]
it also breaks bulk_assign_perm: https://github.com/django-guardian/django-guardian/blob/devel/guardian/managers.py#L70

If the user is inactive, it will still create permissions. If you retry to add permission, it will raise an IntegrityError because of the unique_together.

**Comment by kmmbvnr** [comment]
bulk_create can be fixed by using  `ignore_conflicts=True` flag

The rest django-guardian behaviour matches the Django itself.

List of user permission could be received by temporary setting `.is_active` to a user instance without persisting it into the db

**Comment by thclark** [comment]
> bulk_create can be fixed by using `ignore_conflicts=True` flag
> 
> The rest django-guardian behaviour matches the Django itself.
> 
> List of user permission could be received by temporary setting `.is_active` to a user instance without persisting it into the db

Yes, I think this is the best workaround and anything else is likely to end up in tears in terms of accidentally exposed data
